src/dashboard/auth.ts

```diff
@@ -1,24 +1,27 @@
 import * as bcrypt from 'bcryptjs';
-import { Request, Response, NextFunction } from 'express';
 import { LocalStorage } from 'node-localstorage';
 import * as moment from 'moment';
 
 const localStorage = new LocalStorage('./localstorage/session');
 
 export default class Auth {
-  static isAuthenticated(req: Request, res: Response, next: NextFunction) {
+  static isAuthenticated(req, res, next) {
     const sessionId = req.sessionID;
-    const timestamp = moment(localStorage.getItem(sessionId));
+    const timestamp = localStorage.getItem(sessionId);
 
-    if (moment().diff(timestamp, 'minutes') < 20) {
+    if (timestamp === null) {
+      Auth.logout(sessionId);
+      return res.redirect('/login');
+    }
+
+    if (moment().diff(moment(timestamp), 'minutes') < 20) {
       // Login is valid, refresh timestamp and continue
       Auth.saveTimestamp(sessionId);
-      next();
-    } else {
-      // Not valid anymore, logging out
-      Auth.logout(sessionId);
-      res.redirect('/login');
+      return next();
     }
+    // Not valid anymore, logging out
+    Auth.logout(sessionId);
+    return res.redirect('/login');
```

Comment:
Can you combine the upper and lower logout code for DRY purposes?

Reply:
I can DRY

Reply:
Nice one 👌